src/globus_sdk/experimental/globus_app/globus_app.py

```diff
@@ -220,11 +200,102 @@ def __init__(
         consent_client = AuthClient(app=self, app_scopes=[Scope(AuthScopes.openid)])
         self._validating_token_storage.set_consent_client(consent_client)
 
+    def _resolve_client_info(
+        self,
+        app_name: str,
+        config: GlobusAppConfig,
+        login_client: AuthLoginClient | None,
+        client_id: UUIDLike | None,
+        client_secret: str | None,
+    ) -> tuple[UUIDLike, AuthLoginClient]:
+        """
+        Extracts a client_id and login_client from GlobusApp initialization parameters,
+        validating that the parameters were provided correctly.
+
+        Depending on which parameters were provided, this method will either:
+            1.  Create a new login client from the supplied credentials.
+                *   The actual initialization is performed by the subclass using the
+                    abstract method: _initialize_login_client``.
+            2.  Extract the client_id from a supplied login_client.
+
+        :returns: tuple of client_id and login_client
+        :raises: GlobusSDKUsageError if a single client ID or login client could not be
+            definitively resolved.
+        """
+        if login_client and client_id:
+            msg = "Mutually exclusive parameters: client_id and login_client."
+            raise GlobusSDKUsageError(msg)
+
+        if login_client:
+            # User provided an explicit login client, extract the client_id.
```

Comment:
Any reason why not:
```suggestion
        if login_client:
            if client_id:
                msg = "Mutually exclusive parameters: client_id and login_client."
                raise GlobusSDKUsageError(msg)

            # User provided an explicit login client, extract the client_id.
```

...my preference is not strong, but I was just slightly surprised to see `login_client` checked twice in this way.

Reply:
Just a stylistic choice.

I laid out the code in this pattern in an attempt to establish that neither `login_client` nor `client_id` were the "standard" or "expected" resolution paths. Your suggestion makes it seem a bit more like `login_client` is the standard resolution (mutex checking is a subcase of login_client existence).

My preference is also not strong here. If the check in question had the capability to be expensive at all I'd definitely swap but I can't imagine `bool(login_client)` doing much more than a `None` identity check.

Reply:
I hadn't read it that way, but I'm totally fine with that reasoning!